DEVELOPER_GUIDE.md

```diff
@@ -1,333 +1,279 @@
 
+### `distribution`
 
-> **NOTE:** If you have imported the project into IntelliJ IDEA the project will
-> be automatically configured to add the correct license header to new source
-> files based on the source location.
+Builds our tar and zip archives and our rpm and deb packages.
 
+### `libs`
 
+Libraries used to build other parts of the project. These are meant to be internal rather than general purpose. We have no plans to
+[semver](https://semver.org/) their APIs or accept feature requests for them. We publish them to maven central because they are dependencies of our plugin test framework, high level rest client, and jdbc driver but they really aren't general purpose enough to *belong* in maven central. We're still working out what to do here.
 
-## Running The Full Test Suite
+### `modules`
 
-**Note:  OpenSearch hasn't made any changes to the test suite yet beyond fixing tests that broke after removing non-Apache licensed code and non-Apache licensed code checks.  Also, while we're in pre-alpha, some tests may be failing until we finish the forking process.  We should have an issue for all failing tests, but if you find one first, feel free to open one (and fix it :) ).**
+Features that are shipped with OpenSearch by default but are not built in to the server. We typically separate features from the server because they require permissions that we don't believe *all* of OpenSearch should have or because they depend on libraries that we don't believe *all* of OpenSearch should depend on.
 
+For example, reindex requires the `connect` permission so it can perform reindex-from-remote but we don't believe that the *all* of OpenSearch should have the "connect". For another example, Painless is implemented using antlr4 and asm and we don't believe that *all* of OpenSearch should have access to them.
 
-Before submitting your changes, run the test suite to make sure that nothing is broken, with:
+### `plugins`
 
-```sh
-./gradlew check
-```
-If we're still running down issues, you may want to start with just
+Officially supported plugins to OpenSearch. We decide that a feature should be a plugin rather than shipped as a module because we feel that it is only important to a subset of users, especially if it requires extra dependencies.
 
-```sh
-./gradlew precommit
-```
+The canonical example of this is the ICU analysis plugin. It is important for folks who want the fairly language neutral ICU analyzer but the library to implement the analyzer is 11MB so we don't ship it with OpenSearch by default.
 
+Another example is the `discovery-gce` plugin. It is *vital* to folks running in [GCP](https://cloud.google.com/) but useless otherwise and it depends on a dozen extra jars.
 
-#### `qa`
+### `qa`
 
-Honestly this is kind of in flux and we're not 100% sure where we'll end up.  We welcome your throughts and help.
+Honestly this is kind of in flux and we're not 100% sure where we'll end up. We welcome your throughts and help.
```

Comment:
This existing but I think we can fix it "throughts" -> "thoughts"

Reply:
🤦